src/Analysis/Ast/Impl/Analyzer/AnalysisModuleKey.cs

```diff
@@ -26,28 +26,34 @@ internal struct AnalysisModuleKey : IEquatable<AnalysisModuleKey> {
         public string FilePath { get; }
         public bool IsTypeshed { get; }
 
+        private readonly int _hashCode;
+
         public AnalysisModuleKey(IPythonModule module) {
             Name = module.Name;
             FilePath = module.ModuleType == ModuleType.CompiledBuiltin ? null : module.FilePath;
             IsTypeshed = module is StubPythonModule stub && stub.IsTypeshed;
+            _hashCode = CreateHashCode(Name, FilePath, IsTypeshed);
```

Comment:
you might want to calculate the hash lazily if creating it multiple time was an issue at the first place?

Reply:
ah, sorry didn't see the description that saying HashCode is always get called.

Reply:
Lazily was what I'd have done, but unfortunately this is a struct so it'd need to be done in the initializer (unless there's something I'm missing about struct behaviors).

Reply:
There is no benefit of doing it lazy. This type is made to be a key in the dictionary, so that is a proper fix.